tests/test_singularity.py

```diff
@@ -127,3 +129,20 @@ def test_singularity_docker_image_id_in_tool(tmp_path: Path) -> None:
         ["--singularity", get_data("tests/debian_image_id.cwl"), "--message", "hello"]
     )
     assert result_code1 == 0
+
+
+@needs_singularity_3_or_newer # type: ignore
+def test_singularity_userns_error() -> None:
```

Comment:
```suggestion
def test_singularity_userns_error() -> None:
    """
    Confirm the behaviour of Singularity 3.x --userns in Travis CI.
    
    Travis CI Xenial is not yet running a kernel with userspace namespaces enabled yet.
    """
```

to fix `D103: Missing docstring in public function`

Now that we have documented it, we see that this is not the best test. Our Travis CI setup runs Singularity 2.x, so this function is never run. The test function targets an external program, not the code that was introduced in this PR. And different users will have different experiences based upon their kernel version and configuration.

Since the changes are straight forward, and at least privately tested, I am okay if you remove this test.

Reply:
Got it, just removed the test. Thanks again for the feedback!